packages/fx-core/src/plugins/resource/apim/managers/apimManager.ts

```diff
@@ -207,53 +215,65 @@ export class ApimManager {
     await apimService.addApiToProduct(resourceGroupName, apimServiceName, productId, apiId);
   }
 
-  public async updateArmTemplates(): Promise<ArmTemplateResult> {
+  public async updateArmTemplates(ctx: PluginContext): Promise<ArmTemplateResult> {
+    const azureSolutionSettings = ctx.projectSettings?.solutionSettings as AzureSolutionSettings;
+    const plugins = getActivatedV2ResourcePlugins(azureSolutionSettings).map(
+      (p) => new NamedArmResourcePluginAdaptor(p)
+    ); // This function ensures return result won't be empty
```

Comment:
move the comment to the beginning of code?

Reply:
will resolve in another pr